Framework/Algorithms/src/SumOverlappingTubes.cpp

```diff
@@ -237,17 +221,22 @@ void SumOverlappingTubes::getHeightAxis() {
     const auto &compAss = dynamic_cast<const ICompAssembly &>(*comp);
```

Comment:
Is there a good reason you are not using `ComponentInfo` for this? @LamarMoore is reviewing some changes at the moment that give component info a `indexOfAny(const std::string& name` (but it's pretty much there now anyway`. and would be much cleaner than the implementation here.

Reply:
No dynamic_cast needed for one thing.